actors/runtime/src/runtime/fvm.rs

```diff
@@ -263,28 +261,45 @@ where
         value: TokenAmount,
     ) -> Result<RawBytes, ActorError> {
         if self.in_transaction {
-            return Err(actor_error!(SysErrIllegalActor; "runtime.send() is not allowed"));
+            return Err(actor_error!(user_assertion_failed; "runtime.send() is not allowed"));
         }
-        // TODO: distinguish between "syscall" errors and actor exit codes.
         match fvm::send::send(&to, method, params, value) {
             Ok(ret) => {
                 if ret.exit_code.is_success() {
                     Ok(ret.return_data)
                 } else {
-                    Err(ActorError::from(ret.exit_code))
+                    // The returned code can't be simply propagated as it may be a system exit code.
+                    // TODO: improve propagation once we return a RuntimeError.
+                    // Ref https://github.com/filecoin-project/builtin-actors/issues/144
+                    Err(actor_error!(
+                        user_assertion_failed,
+                        format!(
+                            "send to {} method {} aborted with code {}",
+                            to, method, ret.exit_code
+                        )
+                    ))
                 }
             }
             Err(err) => Err(match err {
+                // Some of these errors are from operations in the Runtime or SDK layer
+                // before or after the underlying VM send syscall.
                 ErrorNumber::NotFound => {
-                    actor_error!(SysErrInvalidReceiver; "receiver not found")
+                    // How do we know the receiver is what triggered this?
+                    // An error number doesn't carry enough information at this level
+                    // above the raw syscall (and even then).
+                    actor_error!(not_found; "receiver not found")
                 }
                 ErrorNumber::InsufficientFunds => {
-                    actor_error!(SysErrInsufficientFunds; "not enough funds")
+                    // Actually this is more like an illegal argument where the caller attempted
+                    // to transfer an amount larger than that representable by the VM's
+                    // token amount type. Yes, the caller doesn't have that amount, but if they'd
+                    // attempted to transfer a representable amount it would fail with
+                    // SYS_INSUFFICIENT_FUNDS instead, so this difference is wierd.
```

Comment:
This isn't correct. If the caller attempts to transfer more funds than they have, they'll get an `InsufficientFunds` syscall error (whether or not the value can be represented). In terms of exit codes:

1. They'll get a `USR_INSUFFICIENT_FUNDS` exit code if the called actor exits with that code for some reason.
2. The caller will _never_ see a `SYS_INSUFFICIENT_FUNDS` exit code. That can only show up in on-chain message receipts from the top-level account.

I've filed https://github.com/filecoin-project/fvm-specs/issues/80 for discussion (not sure if we actually want to do that, but we should make a decision).